src/nvc.c

```diff
@@ -252,6 +252,41 @@ static int elaborate(int argc, char **argv)
          break;
       case 'c':
          opt_set_int(OPT_COVER, 1);
+         if (optarg) {
+            char prev = 0;
+            int n_chars = 0;
+            const char *full_cov_opt = optarg;
+            do {
+               printf("Processing coverage type %c\n", *optarg);
+               if (*optarg == ',' || *optarg == '\0') {
+                  if (prev == 's')
+                     opt_set_int(OPT_COVER_STMT, 1);
+                  else if (prev == 't')
+                     opt_set_int(OPT_COVER_TOGGLE, 1);
+                  else if (prev == 'b')
+                     opt_set_int(OPT_COVER_BRANCH, 1);
+                  else
+                     fatal("Unknown coverage type '%c'. Valid coverage types are: \n"
```

Comment:
Error messages ought to start with a lower case letter to match the style of the rest of the program. If you want to provide additional information you can use the `diag_*` functions like:

```c
diag_t *d = diag_new(DIAG_FATAL, NULL);
diag_printf(d, "unknown coverage type '%c'", *optarg);
diag_hint(d, NULL, "valid coverage types are: \n"
          "  s (statement)\n"
          "  t (toggle)\n"
          "  b (branch)");
diag_hint(d, NULL, "selected coverage types shall be "
          "comma separated e.g $bold$--cover=s,t,b$$");
diag_emit(d);
fatal_exit(EXIT_FAILURE);
```

Reply:
Changed to the code you proposed.